src/util.c

```diff
@@ -2255,6 +2320,24 @@ ni_stringbuf_empty(const ni_stringbuf_t *sb)
 	return sb->len == 0; /* bool */
 }
 
+const char *
+ni_stringbuf_join(ni_stringbuf_t *buf, const ni_string_array_t *nsa, const char *sep)
+{
+	unsigned int i;
+	size_t len;
+
+	if (!buf || !nsa)
+		return NULL;
+
+	len = buf->len;
+	for (i = 0; i < nsa->count; ++i) {
+		if (sep && buf->len)
+			ni_stringbuf_puts(buf, sep);
+		ni_stringbuf_puts(buf, nsa->data[i]);
+	}
+	return buf->string ? buf->string + len : NULL;
```

Comment:
This is different from yours https://github.com/mtomaschewski/wicked/blob/bitmap-utils/src/util.c#L2243 but I found this better.

Reply:
Yes, the changes all make sense.